lib/linebuffer.js

```diff
@@ -34,12 +34,13 @@ const needsRetry = (error, response, body) => {
 
     if (!response) { return false; }
 
-    if (response.statusCode !== 200) {
-        debug(`req failed: ${body}`);
-        if (bufferStats.flushcount % 10 === 0) { utils.log(`request failed: ${body}`); }
+    if (!response.statusCode.toString().startsWith('2')) {
+        debug(`[${response.statusCode}] req failed: ${body}`);
```

Comment:
Does `body` print the string we expect? I think in lots of cases, it only prints as `[object Object]`

Reply:
it wasn't apparently but for making things safer, I need to stringify

Reply:
Updated, @weizou19!

Reply:
I tested on local
```
const obj = {a: 10, b: 30};
`${obj}`
```
It resolves as `[object Object]`